### PR TITLE
Android: Add both a lost and restore phase

### PR DESCRIPTION
**Pull request by unknownbrackets**
It seemed like you may have thought of this at one point.  Here's my theory on what's happening:
1. gl_lost() runs.
2. Some GLLost() creates an object (since it's really "restore".)
3. Some other GLLost() deletes an object (which just happens - unlucky - to be a recently created object.)
4. gl_lost() finishes, but some objects are broken now.

So the goal here is to break those up, and clear/delete things in lost, but recreate on restore.  I don't think it really adds much complexity anywhere.

Could help #8912 or #8906.

-[Unknown]

### CODE REVIEW

**Comment by sum2012** [comment]
#8912 still not fix
log:
https://gist.github.com/sum2012/0f0fb954d6b880658030fcbff03e5c4a

**Comment by sum2012** [comment]
I see compile error on log........

09-11 12:40:52.235: I/NativeApp(28172): gl_lost() restoring 4 items:
09-11 12:40:52.235: I/NativeApp(28172): gl_lost(1 / 4, 0x7f894bcf90, 7bc0de80)
09-11 12:40:52.235: I/NativeApp(28172): Vertex Shader compile error:
09-11 12:40:52.235: I/NativeApp(28172): Fragment Shader compile error:
09-11 12:40:52.235: I/NativeApp(28172): gl_lost(2 / 4, 0x7f7a3b9750, 7bc0de80)
09-11 12:40:52.235: I/NativeApp(28172): Vertex Shader compile error:
09-11 12:40:52.235: I/NativeApp(28172): Fragment Shader compile error:
09-11 12:40:52.235: I/NativeApp(28172): gl_lost(3 / 4, 0x7f7a3022f8, 7bc0df10)

**Comment by unknownbrackets** [comment]
Sorry - try again.  I clearly didn't test enough, because I had a really stupid typo that was causing that.

-[Unknown]

**Comment by sum2012** [comment]
Well done,fix #8912

**Comment by sum2012** [comment]
Can you also remove warning ?
Remove import android.os.PowerManager;

File: Core/Dialog/PSPSaveDialog.cpp [hunk 1]
```diff
@@ -1068,7 +1068,7 @@ void PSPSaveDialog::ExecuteNotVisibleIOAction() {
 		{
 			bool result = param.GetSize(param.GetPspParam());
 			// TODO: According to JPCSP, should test/verify this part but seems edge casey.
-			if (MemoryStick_State() != PSP_MEMORYSTICK_STATE_DRIVER_READY) {
+			if (MemoryStick_State() != PSP_MEMORYSTICK_STATE_INSERTED) {
```
Comment (hrydgard) [inline Core/Dialog/PSPSaveDialog.cpp]: This change seems unrelated?

**Comment by hrydgard** [comment]
I think we've tried this kind of thing before with inconclusive results, but since this helps, I'm totally fine with this as it makes sense.

**Comment by sum2012** [comment]
I think that this fix should mention in readme

**Comment by unknownbrackets** [comment]
@sum2012 okay, sure.

-[Unknown]

File: Core/Dialog/PSPSaveDialog.cpp (hunk 1 above)
Comment (unknownbrackets) [inline Core/Dialog/PSPSaveDialog.cpp]: Sorry, just saw the warning when compiling so snuck it in.

-[Unknown]
